.github/workflows/go.yml

```
@@ -0,0 +1,29 @@
+name: Go
+
+on:
+  push:
+    branches: [ main ]
+  pull_request: {}
+
+jobs:
+  go:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: stable
+
+      - name: Test
+        run: go test -v ./...
+      - name: Vet
+        run: go vet ./...
+      - name: Go linter
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          only-new-issues: true
+
+          # Enable the gosec linter w/o having to create a .golangci.yml config
+          args: -E gosec
+          skip-cache: true
```

```suggestion
jobs:

  go:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Set up Go
        uses: actions/setup-go@v5
        with:
          go-version: stable
          cache: false

      - name: Lint
        uses: golangci/golangci-lint-action@v3
        with:
          version: latest
          only-new-issues: true

          # Enable the gosec linter w/o having to create a .golangci.yml config
          args: -E gosec

      - name: Test
        run: go test -v ./...
```

I quite don't like the newline situation and there's no need to run `go vet` as it's [enabled by default](https://golangci-lint.run/usage/linters/#enabled-by-default) in `golangci-lint`. Also, `golangci-lint` claims to have [superior caching](https://github.com/golangci/golangci-lint-action?tab=readme-ov-file#performance), so let's try that and run tests last.